vmm/src/lib.rs

```diff
@@ -2282,14 +2282,16 @@ impl RequestHandler for Vmm {
                 vm,
                 #[cfg(all(feature = "kvm", target_arch = "x86_64"))]
                 self.hypervisor.clone(),
-                send_data_migration,
+                send_data_migration.clone(),
             )
             .map_err(|migration_err| {
                 error!("Migration failed: {:?}", migration_err);
 
-                // Stop logging dirty pages
-                if let Err(e) = vm.stop_dirty_log() {
-                    return e;
+                // Stop logging dirty pages only for non-local migrations
+                if !send_data_migration.local {
+                    if let Err(e) = vm.stop_dirty_log() {
```

Comment:
How much overhead does the ioctl really add? Hopefully we never started dirty logging in the local case?

Reply:
> How much overhead does the ioctl really add? Hopefully we never started dirty logging in the local case?

The detailed stop_dirty_log() overhead is shown in the comments above.

The code marked here (the first patch) is a fix for the stop_dirty_log() usage defect. Dirty page recording starts with non-local migration, but it may be executed in local migration to stop recording dirty pages. Therefore, we should limit stop_dirty_log() only in the case of non-local migration.